src/Table.tsx

```diff
@@ -250,6 +256,8 @@ export function useDatatableLifecycle({
     labels,
     rowsPerPageOption,
     Components: usedComponents,
+    paginationAlwaysVisible,
+    totalNumberOfResults,
```

Comment:
Very minor: what do you think on doing this?

```suggestion
    shouldShowPagination: paginationAlwaysVisible || totalNumberOfResults > Math.min(rowsPerPageOption || []),
    shouldShowPaginationOptions: paginationAlwaysVisible || totalNumberOfResults > 0,
```

That way, the JSX conditional rendering in the `Table` component will be concise.

Reply:
these extra parameters makes a lot of sense, one question, why do we need (`rowsPerPageOption || []`) this or condition

Reply:
...hmmm. I was using it as a replacement so we don't need to use `Math.min.apply(Math, rowsPerPageOption)`. But I think I was too hasty.

What do we hope to achieve with the `Math.min.apply` again? I might not understand the intention yet.

Reply:
to get the lowest number in the array, because `Math.min` doesn't directly work on arrays

Reply:
Ah! I see. Thanks for explaining.

In that case, isn't the better semantic is to use `rowsPerPage` instead? Because the lowest rows per page option isn't necessarily the current number of rows per page, right?

Reply:
I believe that was for the following scenario:
Let's say we have [5,10,15,20] as options, and 10 is selected but we have 7 results.
So, `rowsPerPage=10` is bigger than `totalNumber`, but I think the options should still be shown, because the user can go and select 5 and get pagination back.

Reply:
Ah, yes! That makes perfect sense. Thanks for explaining!